aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -322,16 +322,6 @@ void shrink_backward_kernel(TensorIteratorBase& iter, const Scalar& value) {
   });
 }
 
-void hardtanh_backward_kernel(TensorIterator& iter, const Scalar& min, const Scalar& max) {
-  AT_DISPATCH_FLOATING_TYPES_AND(at::ScalarType::Half, iter.dtype(), "hardtanh_backward_cuda", [&]() {
-    auto min_val = min.to<scalar_t>();
-    auto max_val = max.to<scalar_t>();
-    gpu_kernel(iter, [min_val, max_val]GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
```

Comment:
If you remove this and instead rely on clamp_backward, the perf will regress significantly (clamp_backward formula is 
```
    auto zero = at::scalar_tensor(0., grad.options());
    return where((self >= *min).logical_and_(self <= *max), grad, zero);
```
 and it requires multiple passes over data, whereas this kernels traverses the data just once.

Reply:
I see. Then is the solution to add a specialized CUDA version of clamp_backward? I don't see clamp_backward has an entry in native_functions.yaml, so I am not sure how to REGISTER_DISPATCH for it.

Reply:
Yeah, probably specialized cuda and cpu versions of clamp_backward are needed (similar to hardtanh_backward that we have). clamp_backward is in FunctionsManual.cpp, and we need to move it to native_functions from there, otherwise we can't have it as a structured kernel? @albanD do we have any policy on which backward functions live exclusively in FunctionsManual, and which are in native_functions? Currently it looks like a mix of both, and probably makes sense to move more things to native_functions? Also we'd need an entry for `clamp_backward` itself in the derivatives.yaml, similar to how we have it for hardtanh_backward.

Reply:
Ho my bad! I though clamp_backward had a custom kernel...
So it is not that much duplicate code then. We just want to route everything to this kernel (optionally renaming it clamp_backward).

There isn't a strict rule. But for CompositeImplicit functions, there is almost no advantage to be in the dispatcher but there is the big disadvantage to have to follow native op FC/BC rules. So we usually keep them in FunctionsManual.
For non-composite implicit stuff where you need either a custom double backward or kernels, we do an op to easily use the codegen.

Reply:
From talking to @albanD offline, it sounds like we should just rename the `hardtanh_backward` native_function op to `clamp_backward` (and then kill the custom `clamp_backward` function in `FunctionsManual.cpp).

Logically, this PR would then be doing 3 things:
- Making `clamp`'s backward more efficient by using `hardtanh_backward` (since its backward is really just the same as hardtanh_backward, but the second is written more efficiently)
- Renaming `hardtanh_backward` -> `clamp_backward`
- Making `hardtanh` composite

Reply:
`clamp_backward` also handles cases where `min` or `max` are not specified, so some care is needed for that.

Reply:
Also `clamp` supports tensor as `min` or `max`, but `hardtanh` only support scalar `min` and `max`. At this point, I feel it's worth to split this PR into, 1. improve `clamp_backward`'s efficiency; 2. make `hardtanh` composite and delegate `hardtanh_backward` to `clamp_backward`.